From fce6a971eba520ab07df7a20ecc2094f93bbac3e Mon Sep 17 00:00:00 2001
From: John Levon <john.levon@joyent.com>
Date: Wed, 16 Oct 2019 16:31:12 +0000
Subject: [PATCH] TRITON-1888 server-setup workflow should always reboot
 Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by: Josh Wilsdon
 <josh@wilsdon.ca>

---
 lib/workflows/server-setup.js | 24 ++++++++++--------------
 package.json                  |  2 +-
 2 files changed, 11 insertions(+), 15 deletions(-)

diff --git a/lib/workflows/server-setup.js b/lib/workflows/server-setup.js
index f2cab70..d4b0250 100644
--- a/lib/workflows/server-setup.js
+++ b/lib/workflows/server-setup.js
@@ -284,8 +284,16 @@ function markServerAsSetup(job, callback) {
     });
 }
 
-// XXX most of this comes from sendRebootMessage in the server-reboot job
-//     should refactor.
+/*
+ * Note that we always reboot the CN even if the hostname param isn't set.  This
+ * is because an un-setup server is in a slightly odd state - without the zones
+ * pool, vmadmd will be disabled.  Worse, joysetup.sh needs to replace /var, and
+ * the way it does that effectively stomps on /var/run, and in particular any
+ * door servers there.  The simplest way to get to a sane state is to reboot.
+ *
+ * XXX most of this comes from sendRebootMessage in the server-reboot job;
+ * should refactor.
+ */
 function rebootServer(job, callback) {
     var cnapiUrl = job.params.cnapi_url;
     var cnapi = restify.createJsonClient({ url: cnapiUrl});
@@ -297,12 +305,6 @@ function rebootServer(job, callback) {
 
     job.log.info('rebootServer');
 
-    if (!job.params.hostname) {
-        job.log.info('No need to reboot, hostname not set');
-        callback();
-        return;
-    }
-
     cnapi.post(urUrl, payload, function (error, req, res) {
         job.log.info('rebootServer (post)');
         if (error) {
@@ -323,12 +325,6 @@ function waitForReboot(job, callback) {
 
     job.log.info('waitForReboot()');
 
-    if (!job.params.hostname) {
-        job.log.info('No need to wait for reboot, hostname not set');
-        callback();
-        return;
-    }
-
     function checkRebooted() {
         cnapi.get(serverUrl, function (err, req, res, server) {
             var cn_state;
diff --git a/package.json b/package.json
index 4e27c9a..9bdfe79 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "Triton Compute Node API",
-  "version": "1.24.3",
+  "version": "1.24.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.17.2 (Apple Git-113)

